traitsui/tests/test_ui.py

```diff
@@ -87,6 +94,7 @@ def test_reset_with_destroy_wx(self):
 
             ui.reset(destroy=True)
 
+            process_cascade_events()
```

Comment:
What I think is happening is that because additions were made to wx `destroy_children` and `destroy_control` to use `wx.CallAfter`, when `toolkit().destroy_children(self.control)` is called in ui.reset this wont occur until after events are processed.  Is that correct?
What this makes me curious about is that this seems to be changing the behavior of calling `ui.reset` on wx, but we don't expect that to cause any other issues correct?
You mentioned on #1145:
"Alternatively, if deferring deletion for wx should cause issues, we could add another destroy_control_later on Toolkit instead, but I guess that is unlikely to be necessary."
So I take it this has already been considered and you believe it shouldn't be a problem.

Reply:
Yes, this is a change of behaviour for Wx, but it brings it closer in-line to what Qt does, so the toolkit-agnostic code should be fine with it.  It is possible that some particular Wx widgets could have problems introduced by this, but it's not likely - problems are _usually_ caused by things not being there when you expect them to be, rather than the other way around, so delaying destruction shouldn't cause issues.